### PR TITLE
Add master LostFile and lost blocks metric

### PR DESCRIPTION
**Pull request by maobaolong**
### What changes are proposed in this pull request?

Add lost file metric to measure the lost file count.

### Why are the changes needed?

Without this pr, we cannot measure the lost file count

### Does this PR introduce any user facing changes?

No

### CODE REVIEW

**Comment by maobaolong** [comment]
The failed test seems not caused by this PR.

File: core/server/master/src/main/java/alluxio/master/file/LostFileDetector.java [hunk 1]
```diff
@@ -42,6 +44,8 @@
   public LostFileDetector(FileSystemMaster fileSystemMaster, InodeTree inodeTree) {
     mFileSystemMaster = fileSystemMaster;
     mInodeTree = inodeTree;
+    MetricsSystem.registerGaugeIfAbsent(MetricKey.MASTER_LOST_FILE_COUNT.getName(),
+        mFileSystemMaster::getLostFiles);
```
Comment (LuQQiu) [inline core/server/master/src/main/java/alluxio/master/file/LostFileDetector.java:48]: getLostFiles will iterate all the lost blocks and get a list of lost file ids, which if register as gauge is very inefficient.
besides, are you trying to use getLostFiles().size() here?
i will prefer to use cached guaged with a relatively long cached period

Comment (yuzhu) [reply core/server/master/src/main/java/alluxio/master/file/LostFileDetector.java:48]: +1

**Comment by maobaolong** [comment]
@yuzhu @LuQQiu Thanks for the review, I push a new commit, use `registerCachedGaugeIfAbsent` instead.

Besides, I add a new metric `Master.LostBlockCount`.

File: core/common/src/main/java/alluxio/metrics/MetricKey.java [hunk 2]
```diff
@@ -348,6 +348,16 @@ public MetricKey build() {
               + "Use this metric to monitor whether your journal is running out of disk space.")
           .setMetricType(MetricType.GAUGE)
           .build();
+  public static final MetricKey MASTER_LOST_FILE_COUNT =
+      new Builder("Master.LostFileCount")
+          .setDescription("Count of lost file")
```
Comment (apc999) [inline core/common/src/main/java/alluxio/metrics/MetricKey.java:353]: grammar:

```suggestion
          .setDescription("Count of lost files")
```

File: core/common/src/main/java/alluxio/metrics/MetricKey.java [hunk 3]
```diff
@@ -348,6 +348,16 @@ public MetricKey build() {
...
+          .setMetricType(MetricType.GAUGE)
+          .build();
+  public static final MetricKey MASTER_LOST_BLOCK_COUNT =
+      new Builder("Master.LostBlockCount")
+          .setDescription("Count of lost block")
```
Comment (apc999) [inline core/common/src/main/java/alluxio/metrics/MetricKey.java:358]:
```suggestion
          .setDescription("Count of lost unique blocks")
```

**Review by apc999 — COMMENTED** [review]
some minor comments left

**Comment by maobaolong** [comment]
@apc999 Thanks for your suggestion, committed them just now, PTAL.

**Review by LuQQiu — APPROVED** [review]
lgtm, thanks for adding these useful metrics

**Comment by yuzhu** [comment]
alluxio-bot, merge this please

**Pull request merged**